src/OpenTelemetry.AutoInstrumentation/Logging/OtelLogging.cs

```diff
@@ -26,17 +29,50 @@ internal static class OtelLogging
     private const string OtelDotnetAutoLogDirectory = "OTEL_DOTNET_AUTO_LOG_DIRECTORY";
     private const string NixDefaultDirectory = "/var/log/opentelemetry/dotnet";
 
-    private static readonly IOtelLogger OtelLogger;
+    private static readonly IDictionary<string, IOtelLogger> OtelLoggers = new Dictionary<string, IOtelLogger>();
+    private static readonly object Lock = new();
 
-    static OtelLogging()
+    /// <summary>
+    /// Returns Logger implementation.
+    /// </summary>
+    /// <returns>Logger</returns>
+    public static IOtelLogger GetLogger()
+    {
+        return GetLogger(string.Empty);
+    }
+
+    /// <summary>
+    /// Returns Logger implementation.
+    /// </summary>
+    /// <param name="suffix">Suffix of the log file.</param>
+    /// <returns>Logger</returns>
+    public static IOtelLogger GetLogger(string suffix)
+    {
+        if (OtelLoggers.ContainsKey(suffix))
+        {
+            return OtelLoggers[suffix];
+        }
+
+        lock (Lock)
```

Comment:
Maybe `ConcurrentDictionary` instead of trying to synchronize dictionary yourself?